### PR TITLE
[Doc]Restructure monitoring docs to support new and legacy internal collectors

### PR DESCRIPTION
**Pull request by karenzone**
This PR starts with the goodness of #11586, pulls in doc architecture changes from #11614, and resolves recently introduced conflicts.

**PREVIEW:** http://logstash_11714.docs-preview.app.elstc.co/guide/en/logstash/master/configuring-logstash.html

### CODE REVIEW

File: docs/static/settings/monitoring-settings.asciidoc [hunk 1]
```diff
@@ -44,40 +44,48 @@ shipping monitoring data.
...
 Controls how often data samples are collected and shipped on the Logstash side.
 Defaults to `10s`. If you modify the collection interval, set the 
-`xpack.monitoring.min_interval_seconds` option in `kibana.yml` to the same value.
+`monitoring.min_interval_seconds` option in `kibana.yml` to the same value.
 
+`monitoring.cluster_uuid`::
+
+The universally unique identifier (UUID) for the monitoring cluster. 
+By default, {ls} identifies and uses the `cluster uuid` from the first
+`ouput-elasticsearch` plugin defined in the pipelines.
```
Comment (andsel) [inline docs/static/settings/monitoring-settings.asciidoc]: I think this <newline> makes the following phrase disconnected from the `monitoring.cluster_uuid` item,  in the rendered docs http://logstash_11714.docs-preview.app.elstc.co/guide/en/logstash/master/monitoring-internal-collection.html#monitoring-collection-settings

File: docs/static/monitoring/collectors.asciidoc [hunk 2]
```diff
@@ -3,11 +3,11 @@
 [[logstash-monitoring-collectors]]
 ===== Collectors
 
-Collectors, as their name implies, collect things. In {monitoring} for Logstash, 
+Collectors, as their name implies, collect things. In monitoring for Logstash, 
 collectors are just <<pipeline,Inputs>> in the same way that ordinary Logstash 
 configurations provide inputs.
 
-Like {monitoring} for {es}, each collector can create zero or more monitoring 
+Like monitoring for {es}, each collector can create zero or more monitoring
```
Comment (robbavey) [inline docs/static/monitoring/collectors.asciidoc:10]: Any reason why the {monitoring} was switched to just monitoring?

File: docs/static/monitoring/collectors.asciidoc [hunk 3]
```diff
@@ -39,9 +39,9 @@ The collection interval can be configured dynamically and you can also disable
 data collection. For more information about the configuration options for the 
 collectors, see <<monitoring-settings>>.
 
-WARNING: Unlike {monitoring} for {es} and {kib}, there is no 
-`xpack.monitoring.collection.enabled` setting on Logstash. You must use the 
-`xpack.monitoring.enabled` setting to enable and disable data collection. 
+WARNING: Unlike for {es} and {kib} monitoring, there is no 
+`pack.monitoring.collection.enabled` setting on Logstash. You must use the
```
Comment (robbavey) [inline docs/static/monitoring/collectors.asciidoc:43]: Typo

File: docs/static/monitoring/monitoring-internal-legacy.asciidoc [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+==== Migrating from legacy internal collection to new internal collection
+
+Migrating from legacy internal collection to new
+<<monitoring-internal-collection,Internal collection>> is straightforward.  Drop
+the `xpack` prefix from your configuration settings. 
+For example, `xpack.monitoring.enabled` is now `monitoring.enabled`.
+See <<monitoring-settings, Monitoring Settings>> for the list of settings.
+
+If you don’t have an Elasticsearch output plugin configured in the pipelines,
+add the setting `monitoring.cluster_uuid` to your logstash.yml.
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-internal-legacy.asciidoc:22]: Do we need to give an explanation of what this should be, and how they can retrieve it?

File: docs/static/monitoring/monitoring-internal-legacy.asciidoc [hunk 5]
```diff
@@ -0,0 +1,152 @@
+[role="xpack"]
+[[monitoring-internal-collection-legacy]]
+=== Collect {ls} monitoring data using legacy internal collectors
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-internal-legacy.asciidoc:3]: Do we have a description of what is meant by 'legacy'?

File: docs/static/monitoring/monitoring-output.asciidoc [hunk 6]
```diff
@@ -20,28 +20,27 @@ Otherwise, monitoring data might be routed in different ways or to different pla
...
 the following default configuration is used:
 
 [source,yaml]
 ---------------------------------------------------
-xpack.monitoring.elasticsearch.hosts: [ "http://localhost:9200" ]
+monitoring.elasticsearch.hosts: [ "http://localhost:9200" ]
 ---------------------------------------------------
 
-All data produced by {monitoring} for Logstash is indexed in the monitoring 
+All data produced by Logstash monitoring is indexed in the monitoring
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-output.asciidoc:31]: Need to verify whether this is still correct

File: docs/static/monitoring/monitoring-output.asciidoc [hunk 7]
```diff
@@ -20,28 +20,27 @@ Otherwise, monitoring data might be routed in different ways or to different pla
...
 steps are necessary to properly configure Logstash. For more information, see 
 <<configuring-logstash>>. 
 
-IMPORTANT: When discussing security relative to the `elasticsearch` output, it
-is critical to remember that all users are managed on the production cluster, 
-which is identified in the `xpack.monitoring.elasticsearch.hosts` setting.
-This is particularly important to remember when you move from development 
-environments to production environments, where you often have dedicated 
-monitoring clusters.
+IMPORTANT: When discussing security relative to the `elasticsearch` output,
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-output.asciidoc:39]: I don't *think* this is still correct as we don't route through production clusters any more

File: docs/static/monitoring/collectors.asciidoc (hunk 2 above)
Comment (karenzone) [reply docs/static/monitoring/collectors.asciidoc:10]: Yes. `{monitoring}` renders as "X-Pack monitoring," and we are trying to remove that terminology (as we can) across the Stack docs.  Headings currently carry the X-Pack badge that links to the licensing pages, so the distinction is still there. 

Screen Shot 2020-04-09 at 11 06 00 AM

File: docs/static/settings/monitoring-settings.asciidoc [hunk 8]
```diff
@@ -44,40 +42,49 @@ shipping monitoring data.
...
 
 Controls how often data samples are collected and shipped on the Logstash side.
 Defaults to `10s`. If you modify the collection interval, set the 
-`xpack.monitoring.min_interval_seconds` option in `kibana.yml` to the same value.
+`monitoring.min_interval_seconds` option in `kibana.yml` to the same value.
+
+`monitoring.cluster_uuid`::
 
+The universally unique identifier (UUID) for the monitoring cluster. 
+By default, {ls} identifies and uses the `cluster uuid` from the first
```
Comment (robbavey) [inline docs/static/settings/monitoring-settings.asciidoc:54]: It uses `cluster_uuid`s from each of the elasticsearch outputs, rather than just the first

**Comment by andsel** [comment]
LGTM

File: docs/static/settings/monitoring-settings.asciidoc [hunk 9]
```diff
@@ -44,40 +42,49 @@ shipping monitoring data.
...
-`xpack.monitoring.elasticsearch.ssl.keystore.path`::
+`monitoring.elasticsearch.ssl.keystore.path`::
 
 Optional settings that provide the paths to the Java keystore (JKS) to validate
 the client’s certificate.
 
-`xpack.monitoring.elasticsearch.ssl.keystore.password`::
+`monitoring.elasticsearch.ssl.keystore.password`::
 
 Optional settings that provide the password to the keystore.
```
Comment (robbavey) [inline docs/static/settings/monitoring-settings.asciidoc:90]: Also need to change `verification_mode`

File: docs/static/settings/monitoring-settings.asciidoc [hunk 10]
```diff
@@ -86,23 +93,29 @@ Optional settings that provide the password to the keystore.
 Option to validate the server’s certificate. Defaults to `certificate`. To
 disable, set to `none`. Disabling this severely compromises security.
 
+`monitoring.elasticsearch.sniffing`::
+
+Finds all {es} cluster nodes and adds them to the hosts list.
+
+Defaults to `false`.
```
Comment (robbavey) [inline docs/static/settings/monitoring-settings.asciidoc:100]: Do you need the newline here? It renders slightly weirdly
Screen Shot 2020-04-09 at 11 31 16 AM

File: docs/static/monitoring/collectors.asciidoc [hunk 11]
```diff
@@ -3,11 +3,11 @@
 [[logstash-monitoring-collectors]]
```
Comment (robbavey) [inline docs/static/monitoring/collectors.asciidoc:3]: I'm still seeing the xpack rendering in the preview
Screen Shot 2020-04-09 at 11 35 08 AM

File: docs/static/monitoring/monitoring-internal-legacy.asciidoc (hunk 4 above)
Comment (karenzone) [reply docs/static/monitoring/monitoring-internal-legacy.asciidoc:22]: Not sure how much we should invest in the legacy component as it is no longer one of the preferred approaches. Let's discuss.

File: docs/static/monitoring/monitoring-internal-legacy.asciidoc (hunk 5 above)
Comment (karenzone) [reply docs/static/monitoring/monitoring-internal-legacy.asciidoc:3]: We have notes at key entry points directing users to one of the other options. What are you recommending? 
 
Screen Shot 2020-04-09 at 3 56 14 PM

File: docs/static/monitoring/collectors.asciidoc (hunk 11 above)
Comment (karenzone) [reply docs/static/monitoring/collectors.asciidoc:3]: Are you speaking of the badge in the banner?  It is supposed to be there for now.  The X-Pack badge is a link that takes users to our licensing page. 

We are removing extra instances of X-Pack from the text.

File: docs/static/monitoring/monitoring-output.asciidoc (hunk 6 above)
Comment (karenzone) [reply docs/static/monitoring/monitoring-output.asciidoc:31]: TBD

File: docs/static/monitoring/monitoring-output.asciidoc (hunk 7 above)
Comment (karenzone) [reply docs/static/monitoring/monitoring-output.asciidoc:39]: TBD

File: docs/static/monitoring/monitoring-internal-legacy.asciidoc (hunk 4 above)
Comment (robbavey) [reply docs/static/monitoring/monitoring-internal-legacy.asciidoc:22]: True, although we could just use the description from the newer mechanism

File: docs/static/settings/monitoring-settings.asciidoc (hunk 9 above)
Comment (karenzone) [reply docs/static/settings/monitoring-settings.asciidoc:90]: TBD

File: docs/static/monitoring/monitoring-internal.asciidoc [hunk 12]
```diff
@@ -89,54 +58,72 @@ can specify the location of the PEM encoded certificate with the
...
+
+Monitoring {ls} with internal collectors uses these components:
+
+* <<logstash-monitoring-collectors,Collectors>>
+* <<logstash-monitoring-output,Output>>
 
+These pieces live outside of the default Logstash pipeline in a dedicated
+monitoring pipeline. This configuration ensures that all data and processing has
+a minimal impact on ordinary Logstash processing. Existing Logstash features,
+such as the <<plugins-outputs-elasticsearch,`elasticsearch` output>>, can be
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-internal.asciidoc:116]: I know this comes from previously written copy - but I'm not sure what it is trying to say. I _think_ it is trying to talk about the reuse of certain features from the elasticsearch output, such as retry, but I'm not 100%

**Comment by robbavey** [comment]
@karenzone - I'm fine with committing this as is, and then work on the fine details in a subsequent PR

**Comment by karenzone** [comment]
This PR is getting crushed under its own weight.  I will merge what we have here, and immediately open a new one to track final comments.

**Pull request merged**